atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -651,13 +675,161 @@ def set_build_image(self):
             raise RuntimeError("Build object is malformed, failed to fetch buildroot image")
 
         if build_kind == 'DockerImage':
-            self.config_kwargs['build_image'] = build_name
+            return build_name
         else:
             raise RuntimeError("Build kind isn't 'DockerImage' but %s" % build_kind)
 
+    def process_image_from(self, image_from):
+        build_image = None
+        imagestream = None
+
+        if image_from['kind'] == 'DockerImage':
+            build_image = image_from['name']
+        elif image_from['kind'] == 'ImageStreamTag':
+            imagestream = image_from['name']
+        else:
+            raise UnknownKindException
+
+        return build_image, imagestream
+
+    def set_build_image(self, platforms):
+        """
+        Overrides build_image for worker, to be same as in orchestrator build
+        """
+        current_platform = platform.processor()
+        orchestrator_platform = current_platform or 'x86_64'
+
+        build_config_name = None
+        build_image = None
+        imagestream = None
+        current_buildimage = self.get_current_buildimage()
+        status = get_build_json().get("status", {})
+
+        # orchestrator platform is same as platform on which we want to built on,
+        # so we can use the same image
+        if platforms == set([orchestrator_platform]):
+            self.build_image_digests[orchestrator_platform] = current_buildimage
+            return
+
+        # BuildConfig exists
+        if 'config' in status:
+            config = status['config']
+            if config['kind'] == 'BuildConfig':
+                build_config_name = config['name']
+            else:
+                raise RuntimeError("Build config type isn't BuildConfig : %s" % config['kind'])
```

Comment:
since set_build_image is still a huge function and status isn't used for anything other than getting the config, I think this can safely be turned into a utility function, ie:
`
try:
    build_config_name = self.get_build_config_name()
except KeyError:
   the current else block
`

Reply:
I'm also not sure if we actually need to perform these checks. If 'config' is set to something other than `BuildConfig` object there's something seriously wrong in OpenShift. 

But anyways, having a utility method that encapsulates this would help in reducing the size of this method. I'd even go further, and just let `get_build_config_name` that Mark suggested raise the RuntimeError with the helpful message. In other words, this whole block turns into:
```python
build_config_name = self.get_build_config_name()  # also fetches the status
```

`get_build_config_name` returns `None` if `config not in status`; or the actual build config name; or raises RuntimeError in case of data error.